lib/SynchronousTransport.php

```diff
@@ -19,21 +19,27 @@ public function post(Environment $environment, $url, $payload)
         $environment->debug('posting to ' . $url, array('body' => $payload));
         $headers = array('Content-Type:application/json');
         if (curl_setopt($ch, CURLOPT_POSTFIELDS, $payload) === false) {
-            $environment->exception(new Exception('failed setting payload'));
             curl_close($ch);
-            return false;
+            $environment->exception(new Exception('failed setting payload'));
```

Comment:
If you have a logger set you will have it written in the log. If it would throw the exception, then there is no reason to have a logger option at all I guess.